src/output-json-alert.c

```diff
@@ -93,12 +93,13 @@ typedef struct JsonAlertLogThread_ {
 
 /* Callback function to pack payload contents from a stream into a buffer
  * so we can report them in JSON output. */
-static int AlertJsonPrintStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
+static int AlertJsonDumpStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
 {
     MemBuffer *payload = (MemBuffer *)data;
 
-    PrintStringsToBuffer(payload->buffer, &payload->offset, payload->size,
-                         buf, buflen);
+    // We can't user Print...ToBuffer functions as we need real raw data.
+    memmove(payload->buffer + payload->offset, buf, buflen);
```

Comment:
If I understand memmove correctly, it is a memcpy with a intermediate step: "The  memmove()  function  copies  n bytes from memory area src to memory area dest.  The memory areas may overlap: copying takes place as though the bytes in src are first copied into a temporary array that does not overlap src or dest, and the bytes are then copied from the temporary array to dest." (from the man page)

Here we don't seem to have an overlap case, so then memcpy would make more sense?